src/julia-parser.scm

```diff
@@ -865,7 +865,7 @@
            (begin
              #;(if (and (number? ex) (= ex 0))
                  (error "juxtaposition with literal \"0\""))
-             `(call * ,ex ,(parse-unary s))))
+             `(call .* ,ex ,(parse-unary s))))
```

Comment:
`2 .* x` for `x::Diagonal` is also broken by this PR; why is breaking `2x` worse?

(Such cases could be fixed by adding specialized `broadcast` methods, of course.)